### ISSUE TITLE
A11y regression due to changed return behavior

### ISSUE THREAD

**Issue by adam3smith** [IQSS/dataverse#12644]
Originally reported in  https://github.com/IQSS/dataverse/issues/7895 and then fixed in https://github.com/IQSS/dataverse/pull/8500 and https://github.com/IQSS/dataverse/pull/12390, we changed the behavior of the return key when entering information in the create/edit metadata form.

While the original behavior (clearing metadata on pressing return) was clearly a bug, I think we should (need to?) reconsider disabling form submission via return:

The edit metadata dialog is long and if you enter a field somewhere in a middle, it can take 40+ tabs to be able to submit the form afterwards -- a simple one-click operation with the mouse. That's bad accessibility and arguably a violation of [WCAG SC 2.1.1](https://www.w3.org/TR/WCAG21/#keyboard) (and at least the spirit of [2.4.1](https://www.w3.org/TR/WCAG21/#bypass-blocks)).
Moreover, [implicit submission](https://html.spec.whatwg.org/#implicit-submission) is an HTML standard, that we're now purposefully breaking.

Disabling implicit submission in CVoc fields as 12390 does still seems like the right choice to me given the problems describe by @qqmyers related to those fields, but I would like us to go back to web standards (and easy keyboard navigation/accessibility) by restoring form submission on return when the cursor is in a regular text field..
